interfaces/repo.go

```diff
@@ -155,13 +157,16 @@ func (r *Repository) AddPlug(plug *Plug) error {
 	if err := i.SanitizePlug(plug); err != nil {
 		return fmt.Errorf("cannot add plug: %v", err)
 	}
-	if _, ok := r.plugs[plug.Snap.Name()][plug.Name]; ok {
-		return fmt.Errorf("cannot add plug, snap %q already has plug %q", plug.Snap.Name(), plug.Name)
+	if _, ok := r.plugs[snapName][plug.Name]; ok {
+		return fmt.Errorf("cannot add plug %q, snap %q already has a plug with that name", plug.Name, snapName)
```

Comment:
From the user perspective, there's no plug or slot being _added._ I suggest reporting this as:
```
"snap %q has plugs conflicting on name %q"
"snap %q has plug and slot conflicting on name "%q"
```
and so on.